feat(curriculum): Add fCC Authors Page Project to JS Cert (#49826)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641da7bfbc7f0f477438ad8a.md

---
id: 641da7bfbc7f0f477438ad8a
title: Step 16
challengeType: 0
dashedName: step-16
---

# --description--

Now it's time to call the `displayAuthors` function. But again, you don't want to populate the page with all the authors at once. Instead, you can extract a portion of the authors with the `startingIndex` and `endingIndex` variables. The best method to do this is the `.slice()` array method.

First, remove the console log statement showing `authorDataArr`. Then, call the `displayAuthors` function with the `authorDataArr` array and `.slice()`. Use the `startingIndex` variable for the starting point and the `endingIndex` variable for the ending point.

# --hints--

You should assign `data` to the `authorDataArr` variable

```js
assert.match(code, /authorDataArr\s*=\s*data;?/)
```

You should remove the console log statement showing `authorDataArr`

```js
assert.notMatch(code, /console\.log\(("|'|`)Author\s+Data\s+Array:\s*\1\,\s*authorDataArr\);?/)
```

You should call your `displayAuthors` function.

```js
assert.match(code, /displayAuthors\(/)
```

> You should call your `displayAuthors` function by chaining the `slice()` method to your `authorDataArr` variable.

Reviewer: I think this hint should reference the passing of an argument. 

```suggestion
You should call your `displayAuthors` function and pass in an argument of `authorDataArr.slice()`.
```

```js
assert.match(code, /displayAuthors\(authorDataArr.slice\(/)
```

The first parameter of your `slice()` method should be `startingIndex`.

```js
assert.match(code, /displayAuthors\(authorDataArr.slice\(startingIndex/)
```

The second parameter of your `slice()` method should be `endingIndex`.

```js

> assert.match(code, /displayAuthors\(authorDataArr.slice\(startingIndex\, endingIndex\)\);?/)

Reviewer: I wasn't able to pass with this code because of a missing space between `startingIndex` and `endingIndex`

```
displayAuthors(authorDataArr.slice(startingIndex,endingIndex))
```
```suggestion
assert.match(code, /displayAuthors\(authorDataArr.slice\(startingIndex\,\s*endingIndex\)\);?/)
```
